src/test/java/bio/terra/app/controller/AdminApiControllerTest.java

```diff
@@ -144,4 +153,100 @@ void testAdminRetrieveSnapshotNotFound() throws Exception {
+        .thenReturn(true);
+    when(datasetService.retrieveDatasetSummary(MODEL_ID))
+        .thenReturn(new DatasetSummaryModel().id(MODEL_ID).inheritSteward(false));
+    when(datasetService.enableInheritSteward(eq(MODEL_ID), any())).thenReturn(jobId);
+    when(jobService.retrieveJob(eq(jobId), any()))
+        .thenReturn(new JobModel().id(jobId).jobStatus(JobModel.JobStatusEnum.RUNNING));
+
+    Boolean inheritSteward = true;
+    String json =
+        mvc.perform(
+                put(ADMIN_INHERIT_STEWARD_ENDPOINT, MODEL_ID)
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .content(TestUtils.mapToJson(inheritSteward)))
+            .andExpect(status().is(202))
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    JobModel model = TestUtils.mapFromJson(json, JobModel.class);
+    assertThat("Job ID is returned", model.getId(), equalTo(jobId));
+  }
+
+  @Test
+  void testAdminInheritStewardDisable() throws Exception {
+    when(iamService.isResourceTypeAdminAuthorized(
+            any(), eq(IamResourceType.DATASET), eq(IamAction.ADMIN_TOGGLE_INHERIT_STEWARD)))
+        .thenReturn(true);
+    when(datasetService.retrieveDatasetSummary(MODEL_ID))
+        .thenReturn(new DatasetSummaryModel().id(MODEL_ID).inheritSteward(true));
+    Boolean inheritSteward = false;
+    mvc.perform(
+            put(ADMIN_INHERIT_STEWARD_ENDPOINT, MODEL_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(TestUtils.mapToJson(inheritSteward)))
+        .andExpect(status().isInternalServerError());
+  }
+
+  @Test
+  void testAdminInheritStewardNotAuthorized() throws Exception {
+    doThrow(FORBIDDEN_EXCEPTION)
+        .when(iamService)
+        .verifyResourceTypeAdminAuthorized(
+            any(), eq(IamResourceType.DATASET), eq(IamAction.ADMIN_TOGGLE_INHERIT_STEWARD));
+    Boolean inheritSteward = true;
+    mvc.perform(
+            put(ADMIN_INHERIT_STEWARD_ENDPOINT, MODEL_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(TestUtils.mapToJson(inheritSteward)))
+        .andExpect(status().isForbidden());
+  }
+
+  @Test
+  void testAdminInheritStewardInvalidId() throws Exception {
+    Boolean inheritSteward = true;
+    mvc.perform(
+            put(ADMIN_INHERIT_STEWARD_ENDPOINT, "not a UUID")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(TestUtils.mapToJson(inheritSteward)))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  void testAdminInheritStewardDatasetNotFound() throws Exception {
+    when(iamService.isResourceTypeAdminAuthorized(
+            any(), eq(IamResourceType.DATASET), eq(IamAction.ADMIN_TOGGLE_INHERIT_STEWARD)))
+        .thenReturn(true);
+    doThrow(new DatasetNotFoundException("Dataset not found for id: " + MODEL_ID))
+        .when(datasetService)
+        .retrieveDatasetSummary(MODEL_ID);
```

Comment:
It works either way, but I think it's better to use the `when().thenThrow()` form if possible
```suggestion
    when(datasetService.retrieveDatasetSummary(MODEL_ID))
        .thenThrow(new DatasetNotFoundException("Dataset not found for id: " + MODEL_ID));
```